src/pages/_document.tsx

```diff
@@ -0,0 +1,36 @@
+import Document, { Html, Head, Main, NextScript, DocumentContext, DocumentInitialProps } from 'next/document';
+
+const newrelic = require('newrelic');
+
+const AppDocument = ({ browserTimingHeader }: { browserTimingHeader: string }) => {
```

Comment:
I think there are some TS helpers for inferring types form  `getInitialProps` so you don't have to maintain it in two places.

Reply:
Not two places, maintain it manually when you add another prop to gIP return statement

Reply:
```
type Props = DocumentProps & Awaited<ReturnType<Awaited<typeof AppDocument.getInitialProps>>>;

function AppDocument({ browserTimingHeader }: Props) {
```